app/models/sale.rb

```diff
@@ -0,0 +1,6 @@
+class Sale < ActiveRecord::Base
+
+  belongs_to :product
+
+  belongs_to :vendor
```

Comment:
These relationships should be reciprocal. So if a sale belongs_to :vendor, then in the vendor model it should say a vendor has_many :sales